src/main/java/com/bnkrll/controller/SessionController.java

```diff
@@ -0,0 +1,37 @@
+package com.bnkrll.controller;
+
+import com.bnkrll.model.Session;
+import com.bnkrll.service.SessionRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/session")
+@Slf4j
+public class SessionController {
+
+    private final SessionRepository sessionRepository;
+
+
+    public SessionController(@Qualifier("sessionRepositorySimulator")
+                                     SessionRepository sessionRepository) {
+        this.sessionRepository = sessionRepository;
+    }
+
+    @PostMapping
+    public void saveSession(@RequestBody Session session){
+        sessionRepository.save(session);
+    }
+
+
+    @GetMapping("/pastsessions")
+    public List<Session> getPreviousSessions(
+            @RequestParam(value = "numOfSessions", defaultValue = "30")
+                    String numOfSessions){
```

Comment:
You should be able to accept this as an integer instead of a String.

Reply:
The change has been implemented

`    @GetMapping
    public List<Session> getPreviousSessions(
            @RequestParam(value = "last", defaultValue = "30")
                    int last){
        List<Session> previousSessions = sessionRepository.getLastSessions(last);
        return previousSessions;
    }`